### DISCUSSION TITLE
Week 15 Report

### DISCUSSION THREAD

**Discussion by derrrobin** [SE-TINF22B6/CookPit#95] [General]
## Weekly Update: Database Restructuring and Clean Code Progress

This week was pretty short with just 4 days. So here you can read everysthing about our weekly progress
### Database Restructuring
- This week, our focus remained on converting our database structure to the 4th Normal Form (4NF). This transformation is crucial for enhancing the efficiency and consistency of our database.
- Additionally, the development effort required for recipe search functionality has been significant, resulting in limited frontend progress this week. Hence, we have fewer updates to share through screenshots.

### Clean Code Progress
Here is a short summary of our CelanCode documment. The whole report you can find here with more examples:
https://github.com/SE-TINF22B6/CookPit/blob/main/Docs/CleanCode.md

- We dedicated considerable attention to our Clean Code initiative. 
  - **Naming Conventions**: Clear naming conventions have been established to ensure consistency, clarity, and ease of understanding throughout the codebase.
    - *Descriptive Names*: Component names are carefully chosen to reflect their roles and functionalities clearly.
    - *Consistency*: We maintain uniform naming patterns to keep the codebase organized and coherent.
    - *Readability*: Prioritization of readability facilitates better comprehension for both existing and new team members.
  - **Commenting and Documentation**: Emphasis was placed on comprehensive code commenting and documentation to enhance readability and usability.
  - **Consistent Code Formatting**: Maintaining consistent code formatting is essential for readability and organization.
  - **Single Responsibility Principle (SRP)**: We adhere to SRP by ensuring each function and method has a specific responsibility, thereby improving code clarity and maintainability.

Here are some Examples shown: 

## Clear naming:

![image](https://github.com/SE-TINF22B6/CookPit/assets/123726577/cd92c6f0-1b4f-4b9c-bd6b-b7f62bda893d)

## Clear folder structure:

![image](https://github.com/SE-TINF22B6/CookPit/assets/123726577/f68c7798-3266-4c8d-be39-e9748fe6061e)

## Clear description

![image](https://github.com/SE-TINF22B6/CookPit/assets/123726577/c024716d-f36a-40e5-986d-3606afbbd068)

That´s all. We hope you liked it, your CookPit Team :)

**Comment by fehlend0** [comment]
Dear CookPit Team,

it's impressive to see your commitment to database optimization and clean code practices. Converting to 4NF is no small feat, and your focus on clear naming conventions and readability will surely pay off in the long run. 

Keep up the excellent work,
your Statify Team

**Comment by Siimonhk** [comment]
Hey There CookPit,

I see that theres a strong focus on technical fundamentals like the addition of Clean code practices. The Conversion to 4NF is surely difficult but beneficial in the end. Impressive work.

**Comment by BenjaminB109** [comment]
Dear CookPit Team,
i think it's very good how much value you place on a well-organized database. Reaching the 4NF could be a lot of work. I also like how you want to create order with the clear descriptions.
